geomstats/geometry/discrete_surfaces.py

```diff
@@ -0,0 +1,313 @@
+            gs.take(point, indices=self.faces[:, 2], axis=-2),
+        )
+        normals_at_point = 0.5 * gs.cross(vertex_1 - vertex_0, vertex_2 - vertex_0)
+        if need_squeeze:
+            normals_at_point = gs.squeeze(normals_at_point, axis=0)
+        return normals_at_point
+
+    def surface_one_forms(self, point):
+        """Compute the vector valued one-forms.
+
+        The one forms are evaluated at the faces of a triangulated surface.
+
+        A one-form is represented by the two vectors (01) and (02) at each face
+        of vertices 0, 1, 2.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n_vertices, 3]
+             Surface, as the 3D coordinates of the vertices of its triangulation.
+
+        Returns
+        -------
+        one_forms_base_point : array-like, shape=[..., n_faces, 2, 3]
+            One form evaluated at each face of the triangulated surface.
+        """
+        vertex_0 = gs.take(point, indices=self.faces[:, 0], axis=-2)
+        vertex_1 = gs.take(point, indices=self.faces[:, 1], axis=-2)
+        vertex_2 = gs.take(point, indices=self.faces[:, 2], axis=-2)
+        one_forms = gs.stack([vertex_1 - vertex_0, vertex_2 - vertex_0], axis=-2)
+        if point.ndim == 3 and one_forms.ndim == 3:
+            one_forms = gs.expand_dims(one_forms, axis=0)
+        return one_forms
+
+    def face_areas(self, point):
+        """Compute the areas for each face of a triangulated surface.
+
+        The corresponds to the volume area for the surface metric, that is
+        the volume area of the pullback metric of the immersion defining the
+        surface metric.
+
+        Parameters
+        ----------
+        point : array-like, shape=[n_vertices, 3]
+            Surface, as the 3D coordinates of the vertices of its triangulation.
+
+        Returns
+        -------
+        _ : array-like, shape=[n_faces,]
+            Area computed at each face of the triangulated surface.
+        """
+        surface_metrics = self.surface_metric_matrices(point)
+        return gs.sqrt(gs.linalg.det(surface_metrics))
+
+    def surface_metric_matrices(self, point):
+        """Compute the surface metric matrices.
+
+        The matrices are evaluated at the faces of a triangulated surface.
+
+        The surface metric is the pullback metric of the immersion q
+        defining the surface, i.e. of
+        the map q: M -> R3, where M is the parameterization manifold.
+
+        Parameters
+        ----------
+        point : array like, shape=[n_vertices, 3]
+            Surface, as the 3D coordinates of the vertices of its triangulation.
+
+        Returns
+        -------
+        metric_mats : array-like, shape=[n_faces, 2, 2]
+            Surface metric matrices evaluated at each face of
+            the triangulated surface.
+        """
+        need_squeeze = False
+        if point.ndim == 2:
+            point = gs.expand_dims(point, 0)
+            need_squeeze = True
```

Comment:
can be removed if handling slightly differently `axes` below.

Reply:
done.